lib/explorer/data_frame.ex

```diff
@@ -1309,28 +1370,30 @@ defmodule Explorer.DataFrame do
       >
   """
   @doc type: :single
-  @spec rename_with(df :: DataFrame.t(), callback :: function(), columns :: list() | function()) ::
+  @spec rename_with(
+          df :: DataFrame.t(),
+          callback :: function(),
+          columns :: :all | columns() | function()
+        ) ::
           DataFrame.t()
-  def rename_with(df, callback, columns \\ [])
+  def rename_with(df, callback, columns \\ :all)
 
-  def rename_with(df, callback, []) when is_function(callback),
+  def rename_with(df, callback, :all) when is_function(callback),
     do: df |> names() |> Enum.map(callback) |> then(&rename(df, &1))
 
-  def rename_with(df, callback, columns) when is_function(callback) and is_list(columns) do
+  def rename_with(df, callback, [column | _] = columns)
+      when is_function(callback) and is_column(column) do
+    columns = to_existing_columns(df, columns)
     old_names = names(df)
 
-    Enum.each(columns, fn name ->
-      maybe_raise_column_not_found(old_names, name)
-    end)
-
     old_names
     |> Enum.map(fn name -> if name in columns, do: callback.(name), else: name end)
     |> then(&rename(df, &1))
   end
 
   def rename_with(df, callback, columns) when is_function(callback) and is_function(columns) do
     case df |> names() |> Enum.filter(columns) do
-      [_ | _] = columns ->
+      [column | _] = columns when is_column(column) ->
         rename_with(df, callback, columns)
 
       [] ->
```

Comment:
Same here, I would not raise if there is nothing to rename. Just return the dataframe as is!